lib/Dialects/VectorExt/Transform/VectorWarpUtils.cpp

```diff
@@ -380,44 +380,35 @@ struct WarpOpTransferWrite : public OpRewritePattern<vector::TransferWriteOp> {
       : OpRewritePattern<vector::TransferWriteOp>(ctx, b),
         distributionMapFn(fn) {}
 
-  LogicalResult matchAndRewrite(vector::TransferWriteOp writeOp,
-                                PatternRewriter &rewriter) const override {
-    auto warpOp = dyn_cast<WarpSingleLaneOp>(writeOp->getParentOp());
-    if (!warpOp)
-      return failure();
-
-    // There must be no op with a side effect after writeOp.
-    Operation *nextOp = writeOp.getOperation();
-    while ((nextOp = nextOp->getNextNode()))
-      if (hasSideEffect(*nextOp))
-        return failure();
-
-    if (!llvm::all_of(writeOp->getOperands(), [&](Value value) {
-          return writeOp.vector() == value ||
-                 warpOp.isDefinedOutsideOfRegion(value);
-        }))
-      return failure();
+  /// Distribute the TransferWriteOp. Only 1D distributions and vector dims that
+  /// are multiples of the distribution ratio are supported at the moment.
+  LogicalResult tryDistributeOp(RewriterBase &rewriter,
+                                vector::TransferWriteOp writeOp,
+                                WarpSingleLaneOp warpOp) const {
     AffineMap map = distributionMapFn(writeOp);
     SmallVector<int64_t> targetShape(writeOp.getVectorType().getShape().begin(),
                                      writeOp.getVectorType().getShape().end());
     assert(map.getNumResults() == 1 &&
            "multi-dim distribution not implemented yet");
     for (unsigned i = 0, e = map.getNumResults(); i < e; i++) {
       unsigned position = map.getDimPosition(i);
+      if (targetShape[position] < distributionRatio)
+        return failure();
```

Comment:
Is this needed? This seems like a subset of the what is tested right below: (`targetShape[position] % distributionRatio != 0`)

Reply:
you're right!